compiler/src/dotty/tools/dotc/core/Contexts.scala

```diff
@@ -262,6 +262,19 @@ object Contexts {
     /** AbstractFile with given path, memoized */
     def getFile(name: String): AbstractFile = getFile(name.toTermName)
 
+    def getSiblingClassfile(tastyFile: AbstractFile): AbstractFile =
+      base.siblingClassfiles.getOrElseUpdate(tastyFile, {
+        val classfile0 = tastyFile.resolveSibling(tastyFile.name.stripSuffix(".tasty") + ".class")
+        if classfile0 == null then
+          val classfile = tastyFile.resolveSibling(tastyFile.name.stripSuffix(".tasty") + "$.class")
```

Comment:
`tastyFile.name.stripSuffix(".tasty")` could be computed once.

Reply:
Are you sure that is always the case? I remember having a module class but no class at some point. Maybe try without it.

Reply:
I pushed another commit without it

Reply:
@nicolasstucki seems to work